frontends/web/public/btcdirect/fiat-to-coin.html

```diff
@@ -10,104 +10,123 @@
-      addresses: {
-        address,
-        currency,
-        id: 'BitBox',
-        walletName: 'BitBox'
-      }
-    });
-
-    btcdirect('set-parameters',
-      mode === 'production' ? {
-        baseCurrency: currency,
-        fixedCurrency: true,
-        quoteCurrency,
-        // paymentMethod: any of 'bancontact', 'bankTransfer', 'creditCard', 'giropay', 'iDeal', 'sofort', 'applepay'
-        showWalletAddress: false,
-      } : {
-        baseCurrency: currency,
-        fixedCurrency: true,
-        paymentMethod: 'sofort', // sandbox currently only supports sofort payment method
-        quoteCurrency,
-        showWalletAddress: false,
+          ? 'https://cdn.btcdirect.eu/fiat-to-coin/fiat-to-coin.js'
+          : 'https://cdn-sandbox.btcdirect.eu/fiat-to-coin/fiat-to-coin.js'
+        );
+
+        btcdirect('init', {
+          token: apiKey,
+          debug: mode === 'debug',
+          locale: window.frameElement?.dataset.locale || 'en-GB',
+          theme: window.frameElement?.dataset.theme || 'light',
+        });
+
+        // fiat to coin order
+        btcdirect('wallet-addresses', {
+          addresses: {
+            address,
+            currency,
+            id: 'BitBox',
+            walletName: 'BitBox'
+          }
+        });
+
+        btcdirect('set-parameters',
+          mode === 'production' ? {
+            baseCurrency: currency,
+            fixedCurrency: true,
+            quoteCurrency,
+            // paymentMethod: any of 'bancontact', 'bankTransfer', 'creditCard', 'giropay', 'iDeal', 'sofort', 'applepay'
+            showWalletAddress: false,
+          } : {
+            baseCurrency: currency,
+            fixedCurrency: true,
+            paymentMethod: 'sofort', // sandbox currently only supports sofort payment method
+            quoteCurrency,
+            showWalletAddress: false,
+          }
+        );
+
+        window.addEventListener('btcdirect-embeddable-fiat-to-coin-order-confirmed', (event) => {
+          console.log('btcdirect-embeddable-fiat-to-coin-order-confirmed', event);
+          // Handle the event
+          // Note that the sent information from the widget is found inside event.detail
+        });
+
+        break;
       }
-    );
+    };
+
+    window.addEventListener('message', onMessage);
+
+    // Request the parent to send attributes
+    window.parent.postMessage({
+      action: 'request-configuration'
+    }, '*');
 
-    window.addEventListener('btcdirect-embeddable-fiat-to-coin-order-confirmed', (event) => {
-      console.log('btcdirect-embeddable-fiat-to-coin-order-confirmed', event);
-      // Handle the event
-      // Note that the sent information from the widget is found inside event.detail
-    });
+    function showError(message) {
+      document.body.append(
+        Object.assign(document.createElement('h1'), {
+          style: 'color: red; padding: 1rem;',
+          textContent: message,
+        })
+      );
+    }
 
   })();
 </script>
```

Comment:
this is the file we host on `shiftcrypto.io` and is already deployed bitboxapp.shiftcrypto.io/widgets/btcdirect/v1/fiat-to-coin.html

but in order for it to communicate with the react app we need the changes in this PR in btcdirect.tsx